segmentstore/server/host/src/test/java/io/pravega/segmentstore/server/host/handler/AppendProcessorTest.java

```diff
@@ -348,6 +364,36 @@ public void testAppendFails() throws Exception {
         verify(mockedRecorder, never()).recordAppend(eq(streamSegmentName), eq(8L), eq(1), any());
     }
 
+    @Test(timeout = 5000)
+    public void testAppendFailChannelClose() throws Exception {
+        String streamSegmentName = "scope/stream/testAppendSegment";
+        UUID clientId = UUID.randomUUID();
+        byte[] data = new byte[]{1, 2, 3, 4, 6, 7, 8, 9};
+
+        // Setup mocks.
+        StreamSegmentStore store = mock(StreamSegmentStore.class);
+        setupGetAttributes(streamSegmentName, clientId, store);
+
+        CompletableFuture<Void> result = new CompletableFuture<>();
+        result.completeExceptionally(new RuntimeException("Fake exception for testing"));
+        when(store.append(streamSegmentName, data, updateEventNumber(clientId, data.length), AppendProcessor.TIMEOUT))
+                .thenReturn(result);
+
+        @Cleanup
+        EmbeddedChannel channel = createChannel(store);
+
+        // Send a setup append WireCommand.
```

Comment:
I'm not being able to map the pattern in the description:

```
connection.send(wirecommand);
connection.close();
```

to here. Could you explain how we are validating the pattern in the PR description?

Reply:
yes, this test verifies only if `connection.close()` is invoked and the netty channel is infact closed.
Will add a new Junit to simulate the exact pattern.

Reply:
This issue was observed when `AuthTokenCheckFailed` wirecommand was not received by the client. Now PR#3887 fixes it. 
![image](https://user-images.githubusercontent.com/11114165/62617703-11d12a00-b930-11e9-8120-21a8af22f8cd.png)

Observed that a similar scenario with `BadAttributeUpdateException` handling in AppendProcessor. I have added a junit to validate that.